nefertari/elasticsearch.py

```diff
@@ -367,6 +374,46 @@ def get_latest_change(es_data):
             yield max(ids[i], key=lambda x: x.creation_time)
 
     @staticmethod
+    def refresh_parent_document(flat_actions, request):
+        if engine.is_object_document(request.context):
+            request.context.refresh()
+            instance = request.context
+        else:
+            response = request.response.json_body
+            item_type = response['_type']
+            item_cls = engine.get_document_cls(item_type)
+            item_id = response[item_cls.pk_field()]
+            instance = item_cls.get_item(**{item_cls.pk_field(): item_id})
+
+        to_refresh = []
+
+        for document, _ in instance.get_parent_documents(nested_only=True):
+            indexable_document = document.to_indexable_dict()
+            to_refresh.append(indexable_document)
+
+        for action in flat_actions:
+            if '_source' not in action:
+                continue
+            for refreshed_document in reversed(to_refresh):
```

Comment:
this used for list clean up.
example:
`
x = [1,2,3]
In [20]: for i in reversed(x):
    ...:     x.remove(i)
    ...:

In [21]: x
Out[21]: []

In [22]: x = [1,2,3]

In [23]: for i in x:
    ...:     x.remove(i)
    ...:

In [24]: x
Out[24]: [2]
`